### DISCUSSION TITLE
i'm finding it tricky to use the b3d renderer via codtype on windows

### DISCUSSION THREAD

**Discussion by jishnuthewalker** [coldtype/coldtype#138] [Q&A] [answered]
[https://github.com/stenson/coldtype-b3d-examples](url) the b3denv is not really working for me, i tried compiling blender bpy as a python module([https://wiki.blender.org/w/index.php?title=Building_Blender/Other/BlenderAsPyModule](url)) but was running into errors... is there no streamlined way to connect blender to coldtype? 

i know there's scripting in st2 but it would be super cool to script 3d animations... it's insane I stumbled acrodd this library and addon I remember being so frustrated not having a decent way to animate and mess with variable type and text in blender in general!

Edit: I had the wrong version of blender installed in the wrong directory... whops...

**Comment by stenson** [comment]
when you say b3denv isn't working for you, what kind of error messages are you seeing? I've used b3denv on windows so it should be possible, though of course there could be bugs.

the coldtype-blender workflow works by running coldtype inside of a running GUI blender instance, rather than by running a headless blender inside of coldtype, so blender-as-a-module won't work.

**Comment by jishnuthewalker** [comment]
Soo i had completely misunderstood the problem... i had a different version of blender installed so I installed blender3.4 and that got b3denv to install like a charm but now when I try to run the coldtype file I get:

`
(venv) PS D:\my_stuff\developer\ColdType\coldtype-b3d-examples-main\coldtype-b3d-examples-main> coldtype -p b3dlo nowin3d.py
    >>> watching... nowin3d.py
>BLENDER> C:\Program Files\Blender Foundation\Blender\blender.exe
Opening blend file: D:\my_stuff\developer\ColdType\coldtype-b3d-examples-main\coldtype-b3d-examples-main\blends\nowin3d.blend...
Traceback (most recent call last):
  File "D:\my_stuff\developer\ColdType\venv\lib\site-packages\coldtype\renderer\__init__.py", line 332, in reload
    self.winmans.b3d.launch(self.source_reader.blender_io())
  File "D:\my_stuff\developer\ColdType\venv\lib\site-packages\coldtype\renderer\winman\blender.py", line 21, in launch
    self.subp = blender_launch_livecode(
  File "D:\my_stuff\developer\ColdType\venv\lib\site-packages\coldtype\blender\render.py", line 46, in blender_launch_livecode
    return subprocess.Popen(args)
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\python\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\python\lib\subprocess.py", line 1440, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

SYNTAX ERROR
FileNotFoundError: [WinError 2] The system cannot find the file specified
`

am I missing something?

**Reply by stenson** [reply, marked as answer]
Ah, I think this may just be an issue with the Coldtype library. `b3denv` is good at finding the path to blender, but I haven't yet integrated that path-finding into Coldtype itself (something I need to do). In the meantime, can you try running `b3denv blender` and then putting what results in a file at the root of your project called `.coldtype.py` (the leading period is important). For instance, on my windows computer, in my `.coldtype.py` file, I have this:

```python
BLENDER_APP_PATH = "C:/Program Files/Blender Foundation/Blender 3.4/blender.exe"
```

Once you have that file saved, Coldtype should be able to find the correct version of Blender to use.

Also, if `b3denv blender` outputs something starting with `/c/`, you'll need to change that to `C:/`

**Reply by jishnuthewalker** [reply]
omg!!! it worked!!! just had to install 'booleanOperations' and 'skia-pathops' using pip in the blender Python interactive console thingy and it worked!!!! thank you so much for such quick responses!

just pasting the commands to install other stuff here for future reference just in case...

`import pip`
`pip.main(['install', 'booleanOperations'])`
`pip.main(['install', 'skia-pathops'])`

coldtype is so cool (pun intended0 seriously, I love it and can't wait to play with this more! gotta touch up my Python basics first lol!
